src/node/audio_buffer_source.rs

```diff
@@ -506,64 +518,108 @@ impl AudioProcessor for AudioBufferSourceRenderer {
                 && self.render_state.buffer_time < buffer_duration
             {
                 let position = self.render_state.buffer_time * sampling_ratio;
-                self.compute_playback_at_position(position, sample_rate);
+                let playhead = position * sample_rate;
+                let playhead_floored = playhead.floor();
+                let prev_index = playhead_floored as usize; // can't be < 0.
+                let k = (playhead - playhead_floored) as f32;
+
+                self.playback_infos[index] = Some((prev_index, k));
```

Comment:
There's something really strange going on numerically, if I render a 3*128 buffer. In the second chunk that is rendered, playback_info contains the index `183` twice!

Reply:
hum that sounds strange indeed, but it would be interesting to see the value of `k` too because it could just be due to floating point errors I guess. I will investigate